File: src/main/java/no/nav/melosys/eessi/models/sed/nav/Pin.java

package no.nav.melosys.eessi.models.sed.nav;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@JsonInclude(Include.NON_NULL)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pin {

    private String identifikator;

    private String land;

    private String sektor;
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/sed/nav/Pin.kt
package no.nav.melosys.eessi.models.sed.nav

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Pin(
    var identifikator: String? = null,
    var land: String? = null,
    var sektor: String? = null,
    var institusjonsid: String? = null,
    var institusjonsnavn: String? = null
) {
    // Fjern nå vi har bruk fra kotlin
    constructor(identifikator: String?, land: String?, sektor: String?) : this(
        identifikator = identifikator,
        land = land,
        sektor = sektor,
        institusjonsid = null,
        institusjonsnavn = null
    )
}
